.github/workflows/deploy-book.yaml

```diff
@@ -6,6 +6,12 @@ on:
     branches:
       - main
   workflow_dispatch:
+  workflow_call:
```

Comment:
I'm not sure I understand the logic here?
I think anything listed under `workflow_call:` will only be executed if called by another workflow with a `uses:` directive.

Is the goal to have `ARM_USERNAME` and `ARM_PASSWORD` stored as environment variables in the runtime environment that calls `jupyter-book build`?

Sorry for snooping around but I'm really interested to see how we can get this working, because I suspect it will become a model for the more general problem of having Cookbooks running on privileged resources!

Reply:
We can remove this section once the upstream secrets are enabled!

Yes - once we get this workflow down, we can add documentation + ensure other cookbooks can use similar secrets

Reply:
This is the reference I was using to set this up https://docs.github.com/en/enterprise-cloud@latest/actions/using-workflows/reusing-workflows